src/ir/possible-contents.h

```diff
@@ -473,6 +473,12 @@ struct TagLocation {
   }
 };
 
+// The location of an exnref materialized by a catch_ref or catch_all_ref clause
+// of a try_table.
+struct CaughtExnRefLocation {
+  bool operator==(const CaughtExnRefLocation& other) const { return true; }
+};
```

Comment:
1. Yes, there is basically a unique source of non-null `exnref`s. This is similar to `getNullLocation`, which creates a unique source of zero values per type.
2. I don't think we can, because at
  https://github.com/WebAssembly/binaryen/blob/cde2a52fc6d679d53d1c5b72b081992d53e01c6b/src/ir/possible-contents.cpp#L2116-L2117
  the values associated to roots are put in an `st::map<Location, PossibleContents>`, without trying to `combine` them. That means each Location can only accommodate a single root value. If we use a `TagLocation` or even the target `BreakLocation` and assign them a root value, we can enter into conflict with other things that would try to put a different root value (such as a null value) for the same location.

Reply:
I think I see now, thanks. So we just need **a** source of content for the places that read a thrown exnref.

Reply:
It's been a while since I read GUFA code so my understanding is bad, but why are all `exnref`s treated the same? All `null`s are actually the same value and can be used interchangeably, but each `exnref` contains a different payload and tag, no?